dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -2294,14 +2592,47 @@ public String calculateExtremeValue(Context context, String valueField,
         maxQuery.setSortField(sortField, sortOrder);
         maxQuery.addSearchField(valueField);
         DiscoverResult maxResult = this.search(context,maxQuery);
-        if (0 < maxResult.getDspaceObjects().size()) {
+        if (0 < maxResult.getIndexableObjects().size()) {
             List<DiscoverResult.SearchDocument> searchDocuments = maxResult
-                .getSearchDocument(maxResult.getDspaceObjects().get(0));
+                .getSearchDocument(maxResult.getIndexableObjects().get(0));
             if (0 < searchDocuments.size() && 0 < searchDocuments.get(0).getSearchFieldValues
                 (valueField).size()) {
                 return searchDocuments.get(0).getSearchFieldValues(valueField).get(0);
             }
         }
         return null;
     }
+
+    /**
+     * Add the necessary fields to the SOLR document to support a Discover Facet on resourcetypename (archived item,
+     * workspace item, workflow item, etc)
+     *
+     * @param document
+     *            the solr document
+     * @param acvalue
+     *            the authority value
+     * @param fvalue
+     *            the human readable value
+     */
+    private void addNamedResourceTypeIndex(SolrInputDocument document, String acvalue, String fvalue) {
+
+        document.addField("namedresourcetype_filter", acvalue);
+
+        String[] avalues = acvalue.split(SolrServiceImpl.AUTHORITY_SEPARATOR);
+        acvalue = avalues[0];
+
+        String avalue = avalues[1];
+        document.addField("namedresourcetype_authority", avalue);
+        document.addField("namedresourcetype_group", avalue);
+        document.addField("namedresourcetype_ac", acvalue);
+
+        Pattern pattern = Pattern.compile(REGEX);
```

Comment:
Since this REGEX is a constant, it's better to compile the `pattern` once and then store that variable as a constant too.

Reply:
I have refactored the code here and the regex is not longer needed